### ISSUE TITLE
Add convenienceType rule 

### ISSUE THREAD

**Issue by facumenzella** [nicklockwood/SwiftFormat#733]
### Why?
There's a rule in Swiftlint called [convenienceType](https://realm.github.io/SwiftLint/convenience_type.html) that we use a lot in our project and does not support autocorrection.

> Types used for hosting only static members should be implemented as a caseless enum to avoid instantiation.

### What?

The idea is to go from this:

```swift
struct Constants {
    static let β = 0, 5
}
```

to this:

```swift
enum Constants {
    static let β = 0, 5
}
```

### Proposal

Add a `convenienceType` rule that takes care of this.

I am happy to take this one ✋

**Comment by nicklockwood** [comment]
Seems like a nice idea 👍

**Comment by nicklockwood** [comment]
Landed in 0.47.0
